src/version.rs

```diff
@@ -0,0 +1,138 @@
+        let patch_str = parts.next();
+        if parts.next().is_none() {
+            return Err("Python version string has too many parts");
+        };
+
+        let major = major_str
+            .parse()
+            .map_err(|_| "Python major version not an integer")?;
+        let (minor, suffix) = split_and_parse_number(minor_str);
+        if suffix.is_some() {
+            assert!(patch_str.is_none());
+            return Ok(PythonVersionInfo {
+                major,
+                minor,
+                patch: 0,
+                suffix,
+            });
+        }
+
+        let (patch, suffix) = patch_str.map(split_and_parse_number).unwrap_or_default();
+        Ok(PythonVersionInfo {
+            major,
+            minor,
+            patch,
+            suffix,
+        })
+    }
+}
+
+impl PartialEq<(u8, u8)> for PythonVersionInfo<'_> {
+    fn eq(&self, other: &(u8, u8)) -> bool {
+        self.major == other.0 && self.minor == other.1
+    }
+}
+
+impl PartialEq<(u8, u8, u8)> for PythonVersionInfo<'_> {
+    fn eq(&self, other: &(u8, u8, u8)) -> bool {
+        self.major == other.0 && self.minor == other.1 && self.patch == other.2
+    }
+}
+
+impl PartialOrd<(u8, u8)> for PythonVersionInfo<'_> {
+    fn partial_cmp(&self, other: &(u8, u8)) -> Option<std::cmp::Ordering> {
+        (self.major, self.minor).partial_cmp(other)
+    }
+}
+
+impl PartialOrd<(u8, u8, u8)> for PythonVersionInfo<'_> {
+    fn partial_cmp(&self, other: &(u8, u8, u8)) -> Option<std::cmp::Ordering> {
+        (self.major, self.minor, self.patch).partial_cmp(other)
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::Python;
+    #[test]
+    fn test_python_version_info() {
+        Python::with_gil(|py| {
+            let version = py.version_info();
+            #[cfg(Py_3_7)]
+            assert!(version >= (3, 7));
+            #[cfg(Py_3_7)]
+            assert!(version >= (3, 7, 0));
+            #[cfg(Py_3_8)]
+            assert!(version >= (3, 8));
+            #[cfg(Py_3_8)]
+            assert!(version >= (3, 8, 0));
+            #[cfg(Py_3_9)]
+            assert!(version >= (3, 9));
+            #[cfg(Py_3_9)]
+            assert!(version >= (3, 9, 0));
+        });
+    }
+
+    #[test]
+    fn test_python_version_info_parse() {
+        assert!(PythonVersionInfo::from_str("3.5.0a1") >= (3, 5, 0));
+        assert!(PythonVersionInfo::from_str("3.5+") >= (3, 5, 0));
+        assert!(PythonVersionInfo::from_str("3.5+") == (3, 5, 0));
+        assert!(PythonVersionInfo::from_str("3.5+") != (3, 5, 1));
+        assert!(PythonVersionInfo::from_str("3.5.2a1+") < (3, 5, 3));
+        assert!(PythonVersionInfo::from_str("3.5.2a1+") == (3, 5, 2));
+        assert!(PythonVersionInfo::from_str("3.5.2a1+") == (3, 5));
+        assert!(PythonVersionInfo::from_str("3.5+") == (3, 5));
+        assert!(PythonVersionInfo::from_str("3.5.2a1+") < (3, 6));
+        assert!(PythonVersionInfo::from_str("3.5.2a1+") > (3, 4));
+    }
+}
```

Comment:
I thought it made sense to move this into its separate file.

Also, does this need to be attached to `Python`? `Py_GetVersion` can be called before the interpreter is even initialized, and returns a pointer to a static.

Reply:
> Also, does this need to be attached to Python? `Py_GetVersion` can be called before the interpreter is even initialized, and returns a pointer to a static.

I wasn't entirely confident on the guarantees, so opted to be safe. If you're confident we can change this to use statics, seems good to me. Maybe as a separate PR to make it easier for users to track patches?

Reply:
> Maybe as a separate PR to make it easier for users to track patches?

Yeah I'd rather do this separately.